### DISCUSSION TITLE
signInWithProvider returns null authorizationCode on Android

### DISCUSSION THREAD

**Discussion by pamafe1976** [firebase/flutterfire#12778] [Q&A]
Hi

I'm using this flutter code to retrieve credential for sign in with apple:

```
      final appleProvider = AppleAuthProvider()
        ..addScope('email')
        ..addScope('name');

      UserCredential credential = await FirebaseAuth.instance.signInWithProvider(appleProvider);
```

In IOS it returns a valid authorizationCode. However in Android the authorizationCode is always null.
I need the authorizationCode to be able to revokeTokenWithAuthorizationCode when the user deletes the account.
What am I doing wrong?

Regards

**Comment by zhouyuePi** [comment]
I ran into the same problem. Were you able to find a solution?

**Reply by zhouyuePi** [reply]
As far as I know, FirebaseAuth.instance.signInWithProvider(appleAuthProvider) only returns authorizationCode on iOS, not on Android. You don't need the authorizationCode if you just need to sign in with Apple on Android. 

However, you do need the authorizationCode if you want to delete the account created by signing in with Apple on Android. In that case, you can use the Sign In With Apple package (https://pub.dev/packages/sign_in_with_apple) to get the authorizationCode. Then you can follow the steps outlined in https://github.com/jooyoungho/apple-token-revoke-in-firebase, but implement the procedure in Flutter. Note you need to get refresh token from Apple and then revoke it, which need to be done in your server.
